demos/common/python/models/yolo.py

```diff
@@ -305,3 +305,98 @@ def sigmoid(x):
                 objects.append(Detection(x - width / 2, y - height / 2, x + width / 2, y + height / 2,
                                          confidence.item(), class_id.item()))
         return objects
+
+
+class YOLOX(Model):
+    def __init__(self, ie, model_path, input_transform, labels=None, threshold=0.5):
+        super().__init__(ie, model_path, input_transform)
+
+        assert len(self.net.input_info) == 1, "Expected 1 input blob"
+        self.image_blob_name = next(iter(self.net.input_info))
+
+        assert len(self.net.outputs) == 1, "Expected 1 output blob"
+        self.output_blob_name = next(iter(self.net.outputs))
+
+        self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        assert self.c == 3, "Expected 3-channel input"
+
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.threshold = threshold
+        self.nms_threshold = 0.65
+        self.expanded_strides = []
+        self.grids = []
+
+    def preprocess(self, inputs):
+        image = inputs
+        resized_image = resize_image(image, (self.w, self.h), keep_aspect_ratio=True)
+
+        padded_image = np.ones((self.h, self.w, 3), dtype=np.uint8) * 114
+        padded_image[: resized_image.shape[0], : resized_image.shape[1]] = resized_image
+
+        meta = {'original_shape': image.shape,
+                'resized_shape': resized_image.shape,
+                'padded_shape': padded_image.shape,
+                'scale': min(self.w / image.shape[1], self.h/image.shape[0])}
```

Comment:
You didn't use the `resized_shape` and `padded_shape` in postprocess. Also the `scale` could be obtained directly in postprocess.

Reply:
Done